### PR TITLE
Add Hornet documentation

### PR DESCRIPTION
**Pull request by JakeSCahill**
# Description of change

Added Hornet documentation, including a migration guide for those coming from IRI.

This PR also includes updates to our terminology. Where we would use IRI node, we now use IOTA node.

Fixes #722 

# Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds new content)

# Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my changes

### CODE REVIEW

**Comment by JakeSCahill** [comment]
A note for reviewers:

If you want to review only Hornet docs, use the **Jump to** dropdown and search `hornet/`. All Hornet docs are in that directory.

File: hornet/1.0/contribute.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+If you want to contribute new documentation or fix an error, see the [contribution guidelines](https://github.com/iotaledger/documentation/blob/develop/.github/CONTRIBUTING.md).
+
+## Share your knowledge
+
+Helping others is an important part of any open source ecosystem.
+
+By sharing your knowledge with others, you can provide a lot of value to the community and maybe inspire someone else to learn and contribute.
+
+Take a look at what discussions are going on in the #iri-discussion channel on [Discord](https://discord.iota.org).
```
Comment (luca-moser) [inline hornet/1.0/contribute.md:27]: Wrong channel

File: hornet/1.0/guides/allowing-remote-pow.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Enabling remote proof of work
```
Comment (luca-moser) [inline hornet/1.0/guides/allowing-remote-pow.md:1]: Maybe this text should proof of work once and then sub sequentially just use "PoW". Right now it is a mix of both.

File: hornet/1.0/guides/migrating-from-iri.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# Migrating from IRI to Hornet
+
+**This topic explains how to migrate to a Hornet node from IRI. It introduces the different options that you have, depending on whether you want to keep the existing transactions in your  ledger.**
```
Comment (luca-moser) [inline hornet/1.0/guides/migrating-from-iri.md:3]:
```suggestion
**This topic explains how to migrate to a Hornet node from IRI. It introduces the different options that you have, depending on whether you want to keep the existing transactions in your ledger.**
```

File: hornet/1.0/guides/migrating-from-iri.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+This tool reads all transactions from the IRI node and broadcasts them to your connected neighbors.
+
+Before using this tool, you must do the following:
+
+- Configure Hornet to start from a global snapshot
+- Prepare IRI for migration
+
+### Configuring Hornet to start from a global snapshot
+
+Hornet will not accept transactions if their timestamps are older than the first milestone in its ledger. Therefore, unless you know that your transactions are younger than the [oldest recorded local snapshot](https://dbfiles.iota.org/?prefix=mainnet/hornet/) (31 May 2020), you should start Hornet from the last global snapshot (July 2018).
```
Comment (luca-moser) [inline hornet/1.0/guides/migrating-from-iri.md:31]:
```suggestion
Hornet will not accept transactions if their timestamps are older than the first milestone in its ledger. Therefore, unless you know that your transactions are younger than the [oldest recorded local snapshot](https://dbfiles.iota.org/?prefix=mainnet/hornet/) (31 May 2020), you should start Hornet from the last global snapshot (July 2019).
```

**Pull request merged**